src/api/app/services/trigger_controller_service/scm_extractor.rb

```diff
@@ -0,0 +1,56 @@
+module TriggerControllerService
+  # NOTE: this class is coupled to GitHub pull requests events and GitLab merge requests events.
+  class ScmExtractor
+    ALLOWED_GITHUB_ACTIONS = ['opened', 'synchronize', 'closed'].freeze
+    ALLOWED_GITLAB_ACTIONS = ['open', 'update', 'close', 'reopen', 'merge'].freeze
+
+    def initialize(scm, event, payload)
+      # TODO: What should we do when the user sends a wwwurlencoded payload? Raise an exception?
```

Comment:
Why would you not allow to fake this? :)

Reply:
What do you mean?

A bit of context: when defining webhooks in GitHub, the user can choose between `application/x-www-form-urlencoded` or `application/json` content types, whereas GitLab only allows `application/json`.

In our user documentation we can encourage the user to always choose `application/json` ... or should we simply accept and handle both?

Reply:
Ah sorry I was not aware of this. I think making json mandatory  in this case it a good compromise. Doesn't code anything...